js/GroupedBarChart.js

```diff
@@ -422,6 +480,98 @@ var GroupedBarChart = function(param)
                 .attr('cy', function (d) { return that.yScale(d.y);})
                 // Turn back to original height
             ;
+
+            that.addZLine = d3.svg.line()
+                .x(
+                    function(d)
+                    {
+                        var x1                  = (_.findWhere(that.x1Scales, { mainCategory : d.x0})).x1Scale;
+                        var subCategoriesLength = (_.findWhere(that.subCategories, {mainCategory : d.x0})).subCategories.length;
+                        var outerPadding        = (that.xScale.rangeBand() - (subCategoriesLength * x1.rangeBand()))/2;
+
+                        return x1.range().length  < that.maxSubCategoryLength ? (outerPadding + that.xScale(d.x0) + x1.rangeBand()/2) : outerPadding + that.xScale(d.x0) + x1.rangeBand()/2;
+                    }
+                )
+                .y(
+                    function(d)
+                    {
+                        return that.zScale(d.z);
+                    }
+                )
+            ;
+
+
+            /*Z data point data*/
+            /*Get all data elemets that have a z value*/
+            var zlineData = _.map(_.filter(data, function(d){ return d.hasOwnProperty("zval");}), function(d){
+                return {
+                    x0: d.xval,
+                    x1: d.name,
+                    z: d.zval
+                };
+            });
+            console.log(zlineData);
+
+            /*Data Point render*/
+            var zLine = that.svg.call(responsivefy)
+                .append("g")
+                .attr("class", "zLine")
+            ;
+
+            /*Z Path*/
+            var zPath = zLine.append("path")
+                .attr("class", "line")
+                .attr('fill', "none")
+                .attr("stroke-width", 2)
+                .attr("d", that.addZLine(zlineData))
+                .data(zlineData)
+            ;
+            /*As an object, this is one solid line, so cannot be multicolored.*/
+            var totalLengthZ = zPath.node().getTotalLength();
+
+            zPath.attr("stroke-dasharray", totalLengthZ + " " + totalLengthZ)
+                .attr("stroke-dashoffset", totalLengthZ)
+                .transition()
+                .delay((barsAnimationTime+circleAniminationTime)*2)
+                .duration(lineanimationTime)
+                .attr("stroke-dashoffset", 0)
+                .attr('stroke', that.colors[that.colors.length-1])
+                .ease("linear")
+                .attr("stroke-width", 2)
+                .attr("stroke-dashoffset", 0)
+            ;
+
+            var zDatapoints = zLine.selectAll("circle")
+                .data(zlineData)
+                .enter().append("circle")
+                //.attr('class', 'dot')
+                .attr('stroke', function(d, i){return "white";})
+                .attr('stroke-width', "0.5")
+                .on('mouseover',c_mouseover)
```

Comment:
There's a bug in both zDataPoint and yDataPoint where if you mouse over the points while they're bouncing, they wont' go back to their original position.

Maybe block the mouse over effect while the points are still bouncing